packages/core/src/indexing-store/historicalStore.ts

```diff
@@ -280,18 +284,13 @@ export const getHistoricalIndexingStore = ({
 
         return decodeRow(row, table, kind);
       } else {
-        const updateRow = encodeRow(update, table, kind);
+        const updateRow = encodeRow({ id, ...update }, table, kind);
 
         const row = await db
           .withSchema(namespaceInfo.userNamespace)
           .insertInto(tableName)
           .values(createRow)
-          .onConflict((oc) =>
-            // If the updateRow is empty, DO UPDATE SET throws a syntax error.
-            Object.keys(updateRow).length === 0
-              ? oc.column("id").doNothing()
-              : oc.column("id").doUpdateSet(updateRow),
-          )
+          .onConflict((oc) => oc.column("id").doUpdateSet(updateRow))
```

Comment:
Including the `id` is better, because if the `DO NOTHING` block hits, the query doesn't return any data - `RETURNING` only works if a row was modified.